app/code/core/Mage/Core/Model/Config.php

```diff
@@ -501,13 +491,44 @@ public function getCache()
     }
 
     /**
-     * Get lock flag cache identifier
+     * Call before building and saving cache to ensure only one process can save the cache
      *
-     * @return string
+     * If failed to get cache lock:
+     *   - CLI: throws exception
+     *   - Other: 503 error
+     *
+     * @return void
+     * @throws Exception
      */
-    protected function _getCacheLockId()
+    public function getCacheSaveLock()
     {
-        return $this->getCacheId() . '.lock';
+        if (!Mage::app()->useCache('config')) {
+            return;
+        }
+        $connection = Mage::getSingleton('core/resource')->getConnection('core_write');
+        $lockWaitTime = getenv('MAGE_CONFIG_CACHE_LOCK_WAIT') ?: (PHP_SAPI === 'cli' ? 60 : 3);
+        if (!$connection->fetchOne("SELECT GET_LOCK('core_config_cache_save_lock', ?)", [$lockWaitTime])) {
```

Comment:
Using `GET_LOCK` feature requires MySQL 5.7+, so we should update requirements section in `README.md` file.

Reply:
Ahh, I thought we already required 5.7, but agreed the README needs to be updated.. Also `GET_LOCK()` is already used in `app/code/core/Mage/Index/Model/Resource/Helper/Mysql4.php`

Reply:
Updated README.md